packages/styled-system/test/styled-components.js

```diff
@@ -51,6 +51,21 @@ describe('styled-components', () => {
     expect(json).toHaveStyleRule('color', 'tomato')
   })
 
+  test('extended intermediate components keep their props with as prop', () => {
+    const StyledBox = styled('div')(space)
+    const Box = props => <StyledBox {...props} />
+    const Card = styled(Box)(color)
```

Comment:
Thanks for this! But this code does look different from the original issue. I *don't* think this sort of extension would work since there's the intermediary `Box` here and `Card` won't pass `space` props through since it's now rendering a `header`, will it? AFAIK, Styled Components can only really pass things through to other Styled Components. Also need to figure out why CI tests aren't running

Reply:
Yes, sorry the code is different.  I didn't realize the difference when I originally created the ticket.  For what it's worth, the issue was still the same...this was working with Styled System v4, but not with v5.

Reply:
Going to close this out since I do not think the `as` prop is intended to be passed through to components that aren't Styled Components